### PR TITLE
delete inactive user's contact info

### PR DESCRIPTION
**Pull request by diegocepedaw**
Once a user is marked inactive, remove his contact information for privacy reasons.

### CODE REVIEW

File: src/oncall/user_sync/ldap_sync.py [hunk 1]
```diff
@@ -382,12 +384,19 @@ def sync(config, engine):
...
+    # mark active and re-add contact information
     logger.debug('Users to reactivate:')
     for username in users_to_reactivate:
         logger.debug(username)
         try:
             engine.execute('UPDATE user SET active = TRUE WHERE name = %s', username)
             stats['users_reactivated'] += 1
+            for key, value in ldap_users[username].items():
+                if value and key in modes:
+                    logger.debug('\t%s -> %s' % (key, value))
```
Comment (jrgp) [inline src/oncall/user_sync/ldap_sync.py]: If we need this log statement, the logging module will do the string interpolation for you. You can replace this with `logger.debug('\t%s -> %s', key, value)`

File: src/oncall/user_sync/ldap_sync.py [hunk 2]
```diff
@@ -382,12 +384,19 @@ def sync(config, engine):
...
+                    user_contact_add_sql = 'INSERT INTO `user_contact` (`user_id`, `mode_id`, `destination`) VALUES (%s, %s, %s)'
+                    engine.execute(user_contact_add_sql, (user_id, modes[key], value))
```
Comment (jrgp) [inline src/oncall/user_sync/ldap_sync.py]: I might be missing something but where is the `user_id` variable defined? I can't find it. Should the user id interpolation should be converted to a subquery that takes in the username instead?

Comment (diegocepedaw) [reply src/oncall/user_sync/ldap_sync.py]: You're not missing anything, I absentmindedly pushed an intermediate set of changes. I've updated the PR with the full changes.

File: src/oncall/user_sync/ldap_sync.py [hunk 3]
```diff
@@ -283,8 +285,13 @@ def sync(config, engine):
         contacts['full_name'] = row.full_name
         contacts['photo_url'] = row.photo_url
 
+    # all oncall users
     oncall_usernames = set(oncall_users)
 
+    # set of inactive oncall users
+    rows = engine.execute('SELECT name FROM user WHERE active = FALSE')
+    inactive_usernames = set((user.name for user in rows))
```
Comment (jrgp) [inline src/oncall/user_sync/ldap_sync.py]: Last comment: the double parens aren't needed if the comprehension is the sole argument for a function call. Eg just `set(user.name for user in rows)` would work.

**Pull request closed without merging**